services/director-v2/src/simcore_service_director_v2/api/dependencies/dynamic_services.py

```diff
@@ -21,14 +21,10 @@ async def get_service_base_url(
 ) -> URL:
 
     # get the service details
-    service_details: RunningServiceDetails = (
+    service_details: RunningDynamicServiceDetails = (
         await director_v0_client.get_running_service_details(node_uuid)
     )
-    # compute service url
-    service_url = URL(
-        f"http://{service_details.service_host}:{service_details.service_port}{service_details.service_basepath}"
-    )
-    return service_url
+    return URL(service_details.legacy_service_url)
```

Comment:
not sure I understand what difference does it make? none... internal_port is mapped to service_port right?

Reply:
No difference. It was using a different mode before. I've migrated to a the correct model and used the renamed properties. Also added a regression test to avoid breaking legacy service compatibility.